sdc/io/csv_ext.py

```diff
@@ -541,84 +518,81 @@ def pandas_read_csv(
     return dataframe
 
 
-def _gen_csv_reader_py_pyarrow(col_names, col_typs, usecols, sep, typingctx, targetctx, parallel, skiprows):
-    func_text, func_name = _gen_csv_reader_py_pyarrow_func_text(col_names, col_typs, usecols, sep, skiprows)
-    csv_reader_py = _gen_csv_reader_py_pyarrow_py_func(func_text, func_name)
-    return _gen_csv_reader_py_pyarrow_jit_func(csv_reader_py)
+def _gen_pandas_read_csv_func_text(col_names, col_typs, py_col_dtypes, usecols, signature=None):
 
+    func_name = 'csv_reader_py'
+    return_columns = usecols if usecols and isinstance(usecols[0], str) else col_names
+
+    column_loc, _, _ = get_structure_maps(col_typs, return_columns)
+    df_type = DataFrameType(
+        tuple(col_typs),
+        types.none,
+        tuple(col_names),
+        column_loc=column_loc
```

Comment:
This implies has_parent=False (so we won't make reflection of changes made to native DF at boxing to python object), but it turns out that we don't support reflection for DFs at all. @AlexanderKalistratov, do you think it's a problem in this context (i.e. we should probably support reflection first)?